File: baekjoon/10844.cpp
#include<iostream>
using namespace std;
int N;
int Dp[101][10];  //길이, 끝나는자리 수

int main() {
	cin >> N;
	Dp[1][0] = 0;
	for (int i = 1; i <= 9; i++) {
		Dp[1][i] = 1;
	}
	for (int i = 2; i <= N; i++) {
		Dp[i][0] = Dp[i - 1][1];
		Dp[i][9] = Dp[i - 1][8];
		for (int j = 1; j <= 8; j++) {
			Dp[i][j] = (Dp[i - 1][j - 1] + Dp[i - 1][j + 1]) % 1000000000;
		}
	}
	int sum = 0;
	for (int i = 0; i <= 9; i++) {
		sum = (sum + Dp[N][i]) % 1000000000;
	}
	cout << sum;
}

File: baekjoon/10844.py
# N=int(input())
# dp=[[] for _ in range(N)]
# dp[0].append(9)
# for i in range(N-1):
#     for j in dp[i]:
#         if j==1:
#             dp[i+1].append(j)
#         else:
#             dp[i+1].append(j-1)
#             dp[i+1].append(j)
# print(dp[N-1])
# print(sum(dp[N-1])%1000000000)

# N=int(input())
# dp=[[] for _ in range(N)]
# dp=[0]*9
# dp[8]=1
# for i in range(N-1):
#     for j in range(8):
#         if j==0:
#             dp[j]+=dp[j]
#         else:
#             dp[j]+=dp[j+1]
# ans=0
# for i in range(9):
#     ans+=dp[i]*(i+1)
# print(ans%1000000000)

#자료참고
n = int(input())
dp = [[0 for i in range(10)] for j in range(101)]
for i in range(1, 10):
    dp[1][i] = 1
for i in range(2, n + 1):
    for j in range(10):
        if j == 0:
            dp[i][j] = dp[i - 1][1]
        elif j == 9:
            dp[i][j] = dp[i - 1][8]
        else:
            dp[i][j] = dp[i - 1][j - 1] + dp[i - 1][j + 1]
print(sum(dp[n]) % 1000000000)